src/main/java/com/linecorp/armeria/server/docs/ThriftDocString.java

```diff
@@ -0,0 +1,160 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+import org.reflections.util.FilterBuilder;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * {@link ThriftDocString} is an DocString extractor for Thrift IDL JSON.
+ * If you want use DocString in {@link DocService}, generate YOUR.json using YOUR.thrift and
+ * latest Thrift compiler with --gen json option. And, place YOUR.json to resources/META-INF/armeria/thrift.
+ * Or, if you set -Darmeria.thrift.json.dir=YOUR/WANT/FOLDER , Armeria will scan
+ * resources/YOUR/WANT/FOLDER/*.json .
+ */
+public final class ThriftDocString {
+
+    private static final String THRIFT_JSON_PATH = getThriftJsonDirectory();
+
+    private static final TypeReference JSON_VALUE_TYPE = new TypeReference<HashMap<String, Object>>() {};
+
+    private static final String FQCN_DELIM = ".";
+
+    private static final String DELIM = "#";
+
+    private static Map<ClassLoader, Map<String, String>> cached = new ConcurrentHashMap<>();
+
+    private ThriftDocString() {
+    }
+
+    /**
+     * Extracts all DocStrings from all Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> getAllDocStrings(ClassLoader classLoader) {
+        return cached.computeIfAbsent(classLoader, loader -> parseDocStrings(loader, getAllThriftJsons(loader)));
+    }
+
+    /**
+     * Parses DocStrings from input Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> parseDocStrings(ClassLoader classLoader, Collection<String> jsonPaths) {
+        final ImmutableMap.Builder<String, String> docStrings = ImmutableMap.builder();
+        for (String jsonPath : jsonPaths) {
+            docStrings.putAll(getDocStringsFromJsonResource(classLoader, jsonPath));
+        }
+        return docStrings.build();
+    }
+
+    private static String getThriftJsonDirectory() {
+        String dir = System.getProperty("armeria.thrift.json.dir", "META-INF/armeria/thrift");
```

Comment:
- Armeria's system properties all start with `com.linecorp.armeria`. How about changing the property name to `com.linecorp.armeria.thrift.jsonDir`?
- Could we log the determined Thrift JSON directory, like we do in:
  - https://github.com/line/armeria/blob/master/src/main/java/com/linecorp/armeria/common/util/NativeLibraries.java#L51
- Could we calculate the Thrift JSON directory once and keep it in a static final field?